conans/test/unittests/client/tools/cppstd_required_test.py

```diff
@@ -0,0 +1,183 @@
+
+    def _create_conanfile(self, compiler, version, os, cppstd, libcxx=None):
+        settings = MockSettings({"arch": "x86_64",
+                                 "build_type": "Debug",
+                                 "os": os,
+                                 "compiler": compiler,
+                                 "compiler.version": version,
+                                 "compiler.cppstd": cppstd})
+        if libcxx:
+            settings.values["compiler.libcxx"] = libcxx
+        conanfile = MockConanfile(settings)
+        return conanfile
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_valid_min_cppstd_from_settings(self, cppstd):
+        """ valid_min_cppstd must accept cppstd less/equal than cppstd in settings
+        """
+        self.assertTrue(valid_min_cppstd(self.conanfile, cppstd, False))
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14"])
+    def test_valid_min_cppstd_from_outdated_settings(self, cppstd):
+        """ valid_min_cppstd returns False when cppstd is greater when supported on settings
+        """
+        self.conanfile.settings.values["compiler.cppstd"] = cppstd
+        self.assertFalse(valid_min_cppstd(self.conanfile, "17", False))
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_valid_min_cppstd_from_settings_with_extension(self, cppstd):
+        """ valid_min_cppstd must returns True when current cppstd in settings has GNU extension and
+            extensions is enabled
+        """
+        with mock.patch("platform.system", mock.MagicMock(return_value="Linux")):
+            with mock.patch.object(OSInfo, '_get_linux_distro_info'):
+                self.conanfile.settings.values["compiler.cppstd"] = "gnu17"
+                self.assertTrue(valid_min_cppstd(self.conanfile, cppstd, True))
+
+                self.conanfile.settings.values["compiler.cppstd"] = "17"
+                self.assertFalse(valid_min_cppstd(self.conanfile, cppstd, True))
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_valid_min_cppstd_from_settings_with_extension_windows(self, cppstd):
+        """ GNU extensions has no effect on Windows for valid_min_cppstd
+        """
+        with mock.patch("platform.system", mock.MagicMock(return_value="Windows")):
+            self.conanfile.settings.values["compiler.cppstd"] = "gnu17"
+            self.assertTrue(valid_min_cppstd(self.conanfile, cppstd, True))
+
+            self.conanfile.settings.values["compiler.cppstd"] = "17"
+            self.assertTrue(valid_min_cppstd(self.conanfile, cppstd, True))
+
+    def test_valid_min_cppstd_unsupported_standard(self):
+        """ valid_min_cppstd must returns False when the compiler does not support a standard
+        """
+        del self.conanfile.settings.values["compiler.cppstd"]
```

Comment:
Do we have to remove `compiler.cppstd`?

Reply:
Yes. As you know, `compiler.cppstd` is optional, so when removing it we are simulating the case where cppstd is not declared by the current profile or command arguments.

The `compiler.cppstd` from arguments or profile will be passed to `conanfile.settings.compiler.cppstd`.